### PR TITLE
fixes content language override fallback #22329

### PR DESCRIPTION
**Pull request by julienV**
Pull Request for Issue #22329

### Summary of Changes
Load the default language override as well

### Testing Instructions
see issue

### Expected result
see issue

### Actual result
see issue

### CODE REVIEW

**Comment by ReLater** [comment]
I have tested this item :white_check_mark: successfully on c3171fb609be1ef5205f34c1076c292630751f91<br /><br />Thank you! The actual override behavior needled me a lot ;-)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22330">issues.joomla.org/tracker/joomla-cms/22330</a>.</sub>

File: libraries/src/Language/Language.php [hunk 1]
```diff
@@ -789,6 +789,28 @@ protected function loadLanguage($fileName, $extension = 'unknown')
 		return $result;
 	}
 
+	/**
+	 * Loads the override strings
+	 *
+	 * @return  void
+	 *
+	 * @since   __deploy_version__
```
Comment (Quy) [inline libraries/src/Language/Language.php:797]: I don't know if it is case sensitive so to be on the safe side change to `__DEPLOY_VERSION__`.

**Comment by infograf768** [comment]
I do understand why this has been proposed, but it is quite arbitrary.

For a monolanguage site
If the string is created for back-end, choosing the site default language would be quite counterproductive. 
Let's not forget that a monolanguage site may have multiple languages installed to provide various UI languagess depending on the user choice.
A site with en-GB and de-DE installed, even if de-DE is the default language for frontend, can be displayed in admin (and site) in en-GB if a registered user have set such his preferences.

Also, this does not work at all on a multilingual site. 
`$this->default` will always be `en-GB`.

I added here the code to var_dump stuff
```
protected function loadOverride()
	{
		$override = array();

		echo "So-called default site language ";
		var_dump($this->default);
		echo "language used on the page ";
		var_dump($this->lang);

		if ($this->lang !== $this->default)
[...]
```

I created a simple override for French site (which is my default site language in a multilingual site containing 3 content languages: fr-FR, it-IT, en-GB) and echoed it in Protostar.
`OVERRIDE_STRING="Mon override special"`

I get when French is used in frontend:
The override IS translated, but en-GB is considered as default.

screen shot 2018-09-23 at 07 28 42


For Italian :
The override is NOT translated. en-GB is considered as default, and language used on the page is at the same time fr-FR (because the override is set in fr-FR and your patch acts such) and it-IT...

screen shot 2018-09-23 at 07 07 39

For English

The override is NOT translated. en-GB is considered as default, and language used on the page is at the same time fr-FR (because the override is set in fr-FR) and your patch acts such) and en-GB...

screen shot 2018-09-23 at 07 13 34

Conclusion:
On a multilingual site, this would work ONLY if the override is created for en-GB site.
same issue in back-end as for monolanguage site

**Comment by infograf768** [comment]
I have tested this item :red_circle: unsuccessfully on c3171fb609be1ef5205f34c1076c292630751f91<br /><br />Set to unsuccessful for the reasons stated above.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/22330">issues.joomla.org/tracker/joomla-cms/22330</a>.</sub>

**Comment by julienV** [comment]
hmmm right, I assumed that default was actually being set to the selected default language... will work on it abit more then

**Comment by infograf768** [comment]
@julienV 
You may want to close this PR until you find another possible solution. In that case it will be easier to make a new PR.

**Comment by julienV** [comment]
ok sure

**Pull request closed without merging**